### PR TITLE
[SPARK-8657] [YARN] Fail to upload conf archive to viewfs

### PR DESCRIPTION
**Pull request by litao-buptsse**
Fail to upload conf archive to viewfs in spark-1.4
JIRA Link: https://issues.apache.org/jira/browse/SPARK-8657

### CODE REVIEW

**Comment by srowen** [comment]
OK to test

**Comment by andrewor14** [comment]
@litao-buptsse would you mind opening this against the master branch so future Spark releases also benefit from the fix?

**Comment by srowen** [comment]
There is also https://github.com/apache/spark/pull/7053 for `master`; the discussion at https://issues.apache.org/jira/browse/SPARK-8657 and the third PR has a little more about why.

**Comment by andrewor14** [comment]
Hm, I'm pretty confused. @litao-buptsse Which is your proposed fix, this one or #7053? Did you intend to have different fixes merged into different branches?

**Comment by srowen** [comment]
I think the code paths are a little different; see discussion at https://github.com/apache/spark/pull/7042

**Comment by litao-buptsse** [comment]
@andrewor14 @srowen #7125 is for master and #7055 is for branch-1.4

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala [hunk 1]
```diff
@@ -912,7 +943,10 @@ object Client extends Logging {
   val ENV_DIST_CLASSPATH = "SPARK_DIST_CLASSPATH"
 
   // Subdirectory where the user's hadoop config files will be placed.
-  val LOCALIZED_HADOOP_CONF_DIR = "__hadoop_conf__"
+  val LOCALIZED_CONF_DIR = "__hadoop_conf__"
```
Comment (tgravescs) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala]: why did you change the name of this?  please revert as previous name was more descriptive and the change is unnecessary.

**Comment by srowen** [comment]
@litao-buptsse this now diverges a lot from its sibling PR at https://github.com/apache/spark/pull/7125 . Did you push changes here unintentionally? this needs to match the other PR.

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala (hunk 1 above)
Comment (litao-buptsse) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala]: OK I got it

**Comment by litao-buptsse** [comment]
@srowen #7125 is for master branch and passed all the test case. You can consider to merge it to master.

But for branch-1.4, not only contains the bugs, but also need some refactor to cache up with the master. My first commit https://github.com/litao-buptsse/spark/commit/cbae84e72cc8c1949e96b0d17c7ff38ca6da7281 is just intend to fix the bugs. The following commits of the #7055 PR is intend to refactor the branch-1.4 to catch up with master branch.

#7125 for master is ready to merge to master. 
But I don't know if I should refactor the branch-1.4? Or it it spark commiter to cherry-pick some commits from master to branch-1.4? 

If no need to do the refactor job, https://github.com/litao-buptsse/spark/commit/cbae84e72cc8c1949e96b0d17c7ff38ca6da7281 can fix the bugs for branch-1.4 and passed all tests.
If I can continue do the branch-1.4 refactor job, I will follow @tgravescs 's advice and review my code again.

**Comment by srowen** [comment]
@litao-buptsse one JIRA = one logical issue, so I'd generally expect PRs for one JIRA to be quite similar when back-ported. However this has a lot more change; are you sure you mean for all of this to be in this PR?

In any event, at this point, your change to `master` ended up very simple and easy to cherry-pick, so you can close this. When it's merged it can be back-ported easily. That's standard procedure.

**Comment by litao-buptsse** [comment]
@srowen OK I will close this PR and remain only #7125 for this issue
